Dockerfile

```diff
@@ -1,32 +1,8 @@
-# Build the manager binary
-FROM golang:1.19 as builder
-ARG TARGETOS
-ARG TARGETARCH
-
-WORKDIR /workspace
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
-
-# Copy the go source
-COPY cmd/main.go cmd/main.go
-COPY internal/controller/ internal/controller/
-
-# Build
-# the GOARCH has not a default value to allow the binary be built according to the host where the command
-# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
-# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
-# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
-RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
-
-# Use distroless as minimal base image to package the manager binary
+# Use distroless as minimal base image to package the logging-operator binary
 # Refer to https://github.com/GoogleContainerTools/distroless for more details
 FROM gcr.io/distroless/static:nonroot
 WORKDIR /
-COPY --from=builder /workspace/manager .
+ADD bin/logging-operator .
```

Comment:
Wouldn't it be better to keep a 2 stage Dockerfile with the first stage building the binary ? Otherwise does it means that the user will have to manually generate the binary with the makefile ?

Reply:
My point is:
- in the CI we've just built the binary before. Looks like a waste of resources to build it again.
- during dev we will probably mostly use the locally-built binary, not the docker image.
- if ever we want to use the image, we need to run `make` before to generate the binary, right.
- The Dockerfile seemed sketchy, trying to add only some specific go files. Looked like a recipe for broken Dockerfile to me, and some extra maintenance.

I agree in some cases it could be cool that the dockerfile builds the binary, but since the main build machinery is done with `make` it didn't make sense to me to have too many different build processes.

Reply:
Already with architect and make doing different stuff, that's too many build processes for me :rofl:

Reply:
Ok that makes sense, especialy since the CI builds the binary

Reply:
I thought it made sense...
But as local `make` does stuff differently than circleci, either I keep adapting Makefile or I re-set Dockerfile to build the binary.
Any solution looks a bit ugly, I'm going for fixing Dockerfile :cry:

Reply:
<img src="https://media4.giphy.com/media/vQqeT3AYg8S5O/giphy.gif"/>